### PR TITLE
WIP: Add support for teams

### PR DESCRIPTION
**Pull request by transitive-bullshit**
- [x] Add support for new endpoints and `teamId` query param to `saasify-client`
- [x] Add support for `teams` and `switch` commands to `saasify-cli`
- [ ] Add basic docs
- [x] Test, test, test...

Could probably use more testing and docs, but the core functionality all seems to be working as intended.

### CODE REVIEW

File: packages/saasify-cli/lib/cli.js [hunk 1]
```diff
@@ -23,6 +23,8 @@ module.exports = async (argv, opts = { }) => {
     .option('-c, --config <path>', 'Path to `saasify.json` file (defaults to cwd)')
     .option('-C, --no-clipboard', 'Do not attempt to copy URL to clipboard')
 
+  // TODO: add ability to set client.teamId in global program options
```
Comment (chrisvxd) [inline packages/saasify-cli/lib/cli.js]: Captured?

File: packages/saasify-cli/lib/auth.js [hunk 2]
```diff
@@ -29,9 +33,23 @@ exports.get = () => {
 exports.signin = ({ token, user }) => {
   conf.set(keyToken, token)
   conf.set(keyUser, user)
+  conf.delete(keyTeamId)
+  conf.delete(keyTeamSlug)
```
Comment (chrisvxd) [inline packages/saasify-cli/lib/auth.js]: Shouldn't these be `set`?

File: packages/saasify-cli/readme.md [hunk 3]
```diff
@@ -56,15 +56,16 @@ Commands:
   dev [options] [path]                                 Starts a local dev server for debugging your deployments
   help [cmd]                                           Displays usage info for [cmd]
   init [options] [project-name]                        Creates a new project based on a template
-  ls|list [project]                                    Lists deployments
+  ls|list [options] [project]                          Lists deployments
   logs [options] <url|deployment>                      Prints the logs for a given deployment
   publish <deploymentId|deploymentUrl>                 Creates a subscription to a project (requires a valid billing source)
   rm|remove [options] [deploymentId|deploymentUrl...]  Removes deployments
-  secrets <command> [name] [value]                     Manages your secret environment variables
+  secrets <command> [name] [value]                      Manages your secret environment variables
```
[the remark this answers is not in the document]
Comment (transitive-bullshit) [reply packages/saasify-cli/readme.md]: good catch :)

File: packages/saasify-cli/lib/cli.js (hunk 1 above)
Comment (transitive-bullshit) [reply packages/saasify-cli/lib/cli.js]: see my other comment about TODOs. agreed we should capture them, but there are so many and most of them i'd deem rally low priority like this one.

File: packages/saasify-cli/lib/auth.js (hunk 2 above)
Comment (transitive-bullshit) [reply packages/saasify-cli/lib/auth.js]: both should work

Comment (transitive-bullshit) [reply packages/saasify-cli/lib/auth.js]: (tested this locally to make sure)

**Pull request merged**